components/brave_wallet_ui/page/screens/composer_ui/select_token_modal/select_token_modal.tsx

```diff
@@ -167,270 +175,362 @@ export const SelectTokenModal = React.forwardRef<HTMLDivElement, Props>(
+        return tokensFilteredByNetwork
+      }
+      // Since LOCALHOST's chainId is shared between coinType's
+      // this check will make sure we are returning the correct
+      // LOCALHOST asset for each account.
+      const coinName = CoinTypesMap[selectedAccountFilter.accountId.coin]
+      const localHostCoins = tokensFilteredByNetwork.filter(
+        (token) => token.asset.chainId === BraveWallet.LOCALHOST_CHAIN_ID
+      )
+      const accountsLocalHost = localHostCoins.find(
+        (token) => token.asset.symbol.toUpperCase() === coinName
+      )
+
+      const chainList = filterNetworksForAccount(
+        networks,
+        selectedAccountFilter.accountId
+      ).map((network) => network.chainId)
+
+      const list = tokensFilteredByNetwork.filter(
+        (token) =>
+          token.asset.chainId !== BraveWallet.LOCALHOST_CHAIN_ID &&
+          chainList.includes(token?.asset?.chainId ?? '')
+      )
+
+      if (
+        accountsLocalHost &&
+        selectedAccountFilter.accountId.keyringId !==
+          BraveWallet.KeyringId.kFilecoin
+      ) {
+        list.push(accountsLocalHost)
+        return list
+      }
+
+      return list.filter((token) => token.asset.visible)
+    }, [networks, selectedAccountFilter, tokensFilteredByNetwork])
+
+    const tokensBySearchValue = React.useMemo(() => {
+      if (searchValue === '') {
+        return tokensFilteredByAccount
+      }
+      return tokensFilteredByAccount.filter(
+        (token) =>
+          token.asset.name.toLowerCase() === searchValue.toLowerCase() ||
+          token.asset.name
+            .toLowerCase()
+            .startsWith(searchValue.toLowerCase()) ||
+          token.asset.symbol.toLocaleLowerCase() ===
+            searchValue.toLowerCase() ||
+          token.asset.symbol
+            .toLowerCase()
+            .startsWith(searchValue.toLowerCase()) ||
+          token.asset.contractAddress.toLocaleLowerCase() ===
+            searchValue.toLowerCase()
+      )
+    }, [tokensFilteredByAccount, searchValue])
+
+    const firstNoBalanceTokenKey = React.useMemo(() => {
+      const token = tokensBySearchValue.filter((token) =>
+        new Amount(token.assetBalance).isZero()
+      )[0]
+      return token ? getAssetIdKey(token.asset) : ''
+    }, [tokensBySearchValue])
+
+    const accountsForPendingSelectedAsset = React.useMemo(() => {
+      if (!pendingSelectedAsset) {
+        return []
+      }
+      return accounts
+        .filter(
+          (account) => account.accountId.coin === pendingSelectedAsset.coin
+        )
+        .sort(function (a, b) {
+          const aBalance = new Amount(
+            getBalance(a.accountId, pendingSelectedAsset, tokenBalancesRegistry)
           )
+          const bBalance = new Amount(
+            getBalance(b.accountId, pendingSelectedAsset, tokenBalancesRegistry)
+          )
+          return bBalance.minus(aBalance).toNumber()
```

Comment:
perf: can we avoid creating 2 Amount objects?


```suggestion
          return new Amount(
            getBalance(b.accountId, pendingSelectedAsset, tokenBalancesRegistry)
          ).minus(getBalance(a.accountId, pendingSelectedAsset, tokenBalancesRegistry)).toNumber()
```

Reply:
Done!